middleware/tests/admin/test_sgx_authorization.py

```diff
@@ -0,0 +1,333 @@
+    def setUp(self):
+        # Sample valid DER signatures
+        self.sigs = [
+            "3045022100f31ee73e3b10c5d610d9f5501e12ce1f2fd31182d0630c8e0db75fba3f35bbe3022056d0703a27937aec36a0a05bd5b85de6144279ab3a66faf266378ce42a838831",  # noqa E501
+            "3046022100d2e039915b4decd3d32d613bdcfc84090560e0e714284ff4c3b454b563d81c7c022100ff0de20f22f75a87cf546a6e3dd9dada082b0bdd01862e1c566e5bdd67f0c3b1",  # noqa E501
+        ]
+
+        # Sample mrenclave values (32-byte hex strings)
+        self.exporter_mrenclave = "aa" * 32
+        self.importer_mrenclave = "bb" * 32
+
+        # Create migration spec
+        self.migration_spec = SGXMigrationSpec({
+            "exporter": self.exporter_mrenclave,
+            "importer": self.importer_mrenclave
+        })
+
+        # Create SGX authorization instance
+        self.sa = SGXMigrationAuthorization(self.migration_spec, self.sigs)
+
+    def test_migration_spec_n_signatures(self):
+        # Test basic property getters and verify signatures list is copied
+        self.assertEqual(self.sa.migration_spec.exporter, self.exporter_mrenclave)
+        self.assertEqual(self.sa.migration_spec.importer, self.importer_mrenclave)
+        self.assertEqual(self.sa.signatures, self.sigs)
+        self.assertIsNot(self.sa.signatures, self.sigs)  # Verify list is copied
+
+    def test_invalid_migration_spec(self):
+        # Test constructor with invalid migration spec
+        with self.assertRaises(ValueError):
+            SGXMigrationAuthorization("not-a-migration-spec", self.sigs)
+
+    def test_invalid_signatures(self):
+        # Test constructor with invalid signatures (non-list)
+        with self.assertRaises(ValueError):
+            SGXMigrationAuthorization(self.migration_spec, "not-an-array")
+
+    def test_invalid_signature(self):
+        # Test constructor with invalid signature format
+        with self.assertRaises(ValueError):
+            SGXMigrationAuthorization(
+                self.migration_spec,
+                [self.sigs[0], "not-a-valid-signature"]
+            )
+
+    def test_to_dict(self):
+        # Test dictionary conversion
+        expected_dict = {
+            "version": 1,
+            "hashes": {
+                "exporter": self.exporter_mrenclave,
+                "importer": self.importer_mrenclave,
+            },
+            "signatures": self.sigs
+        }
+        self.assertEqual(expected_dict, self.sa.to_dict())
+
+    def test_add_signature(self):
+        # Test adding a valid signature
+        new_sig = "3045022100d2dac5b641d6a454cacdff045ab428bfc4c86e"\
+                  "004ff69728050a33788f6e9e7602207e8b3536a7a50185e2"\
+                  "7219237358823b98678fe32aa7a30b31155cbffad3747d"
+        self.sa.add_signature(new_sig)
+        self.assertEqual(self.sa.signatures, self.sigs + [new_sig])
+
+    def test_add_duplicate_signature(self):
+        # Adding the same signature should be allowed
```

Comment:
Why is this allowed?

Reply:
Good question, not sure if it should TBH. I just reproduced signapp's behavior.

We can either modify the logic to turn this into an error or just remove this test case if it's not a big deal, what do you think?

Reply:
As aligned offline, this behavior will be changed so that we don't allow duplicated signatures. The test case will be updated accordingly.